Public/Src/Tools/Tool.MsBuildGraphBuilder/PropertyTrackingLogger.cs

```diff
@@ -0,0 +1,77 @@
+using System.Linq;
+using BuildXL.Utilities.Collections;
+using Microsoft.Build.Framework;
+
+namespace ProjectGraphBuilder
+{
+    internal class PropertyTrackingLogger : ILogger
+    {
+        private readonly HashSet<string> _variablesRead = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+        private bool _nonTrackingSdkResolversExist = false;
+
+        public IEnumerable<string> PotentialEnvironmentVariablesReads
+        {
+            get
+            {
+                // Return a copy of what's there to minimize threading issues.
+                lock (_variablesRead)
+                {
+                    return _variablesRead.ToArray();
+                }
+            }
+        }
+
+        public void Initialize(IEventSource eventSource)
+        {
+            eventSource.AnyEventRaised += (sender, args) =>
+                                          {
+                                              if (args is EnvironmentVariableReadEventArgs evrArgs)
+                                              {
+                                                  lock(_variablesRead)
+                                                  {
+                                                      _variablesRead.Add(evrArgs.EnvironmentVariableName);
+                                                  }
+                                              }
+                                              else if (args is UninitializedPropertyReadEventArgs upArgs)
+                                              {
+                                                  lock(_variablesRead)
+                                                  {
+                                                      _variablesRead.Add(upArgs.PropertyName);
+                                                  }
+                                              }
+                                              else if (args is BuildWarningEventArgs warnArgs && warnArgs.Code == "MSB4263")
+                                              {
+                                                  // We just want to know whether or not we can trust the above tracking data.
+                                                  _nonTrackingSdkResolversExist = true;
+                                              }
+                                          };
+        }
+
+        public void Shutdown()
+        {
+        }
+
+        public LoggerVerbosity Verbosity { get; set; }
+
+        public string Parameters { get; set; }
+
+        public IReadOnlyCollection<string> BuildAffectingEnvironmentVariables
+        {
+            get
+            {
+                IEnumerable<string> copy;
+
+                lock(_variablesRead)
+                {
+                    copy = _variablesRead.ToArray();
+                }
+
+                return copy.AsReadOnlyCollection();
+            }
+        }
+
+        public bool NonTrackingSdkResolversExist => _nonTrackingSdkResolversExist;

```

Comment:
>NonTrackingSdkResolversExist [](start = 20, length = 28)

Nit: I'd just call this something like 'IsTrackingAvailable'. The Sdk resolver thing sounds like a detail at this level

Reply:
The property is removed in the latest commit. It's possible there could be many reasons tracking isn't working/available/reliable. Sdk resolution is just one. Regardless, they'll all be reported as failures to the graph builder.